# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1354926859 18000
#      Fri Dec 07 19:34:19 2012 -0500
# Node ID 72968fb32c82b3f093408220a09f8f9cdd9f70ad
# Parent  05c781cca57e0cc04f120e859525333f445b90bc
Avoid future timestamp warning on non-existent shared libs

* liboctave/util/oct-shlib.cc (octave_shlib::shlib_rep): Check file
existence before comparing timestamp.

diff --git a/liboctave/util/oct-shlib.cc b/liboctave/util/oct-shlib.cc
--- a/liboctave/util/oct-shlib.cc
+++ b/liboctave/util/oct-shlib.cc
@@ -70,25 +70,25 @@ octave_shlib::shlib_rep::shlib_rep (cons
       ("Octave:warn-future-time-stamp",
        "timestamp on file %s is in the future", file.c_str ());
 }
 
 bool
 octave_shlib::shlib_rep::is_out_of_date (void) const
 {
   file_stat fs (file);
-  return fs.is_newer (tm_loaded);
+  return (fs && fs.is_newer (tm_loaded));
 }
 
 void
 octave_shlib::shlib_rep::fake_reload (void)
 {
   // We can't actually reload the library, but we'll pretend we did.
   file_stat fs (file);
-  if (fs.is_newer (tm_loaded))
+  if (fs && fs.is_newer (tm_loaded))
     {
       tm_loaded = fs.mtime ();
 
       (*current_liboctave_warning_handler)
         ("library %s not reloaded due to existing references", file.c_str ());
     }
 }
 
